flytekit/annotated/python_function_task.py

```diff
@@ -167,9 +174,17 @@ def __init__(
             **kwargs,
         )
         self._task_function = task_function
+        self._execution_mode = execution_mode if execution_mode is not None else self.ExecutionMode.DEFAULT
```

Comment:
this has been bugging me to. Can we just start with an ExecutionMode always set to something like DEFAULT?

Reply:
done